tests/trainer/loops/test_evaluation_loop.py

```diff
@@ -101,3 +102,30 @@ def validation_step(self, batch, batch_idx):
     torch.cuda.empty_cache()
     trainer = Trainer(gpus=1, default_root_dir=tmpdir, fast_dev_run=2, move_metrics_to_cpu=True, weights_summary=None)
     trainer.fit(BoringLargeBatchModel())
+
+
+def test_evaluation_loop_doesnt_store_outputs_if_epoch_end_not_overridden(tmpdir):
+    did_assert = False
+
+    class TestModel(BoringModel):
+        def on_test_batch_end(self, outputs, *_):
+            # check `test_step` returns something
+            assert outputs is not None
+
+    class TestLoop(EvaluationEpochLoop):
+        def on_advance_end(self):
+            # should be empty
+            assert not self.outputs
+            # sanity check
+            nonlocal did_assert
+            did_assert = True
+            super().on_advance_end()
+
+    model = TestModel()
+    # make sure this hook is not overridden
+    model.test_epoch_end = LightningModule.test_epoch_end
```

Comment:
noob question: Does this work if set to None ?

Reply:
Actually it does, because `is_overridden` returns `False` in that case. Changed the test to check this